### ISSUE TITLE
Session Headers Repeated

### ISSUE THREAD

**Issue by DisabledMonkey** [salesagility/SuiteCRM-Core#509]
### Issue

The `PHPSESSID` and `LEGACYSESSID` cookies are repeated a crazy number of times in the response headers
<img width="711" alt="image" src="https://github.com/user-attachments/assets/cbee58d3-3b10-4fce-9be7-87d4c90427d0">

This causes problems when hosting SuiteCrm behind a proxy such as nginx that has default max sizes allowed for the response headers.

### Possible Fix

Make sure `PHPSESSID` and `LEGACYSESSID`  are only returned once in the headers like would be expected

### Steps to Reproduce the Issue

```bash
1. Make a request
2. Look at chrome devtools and see all the headers that are returned.
```

### Context

We run suitecrm on kubernetes which proxies via NGINX Gateway Fabric
Currently you can't [configure it](https://www.github.com/nginxinc/nginx-gateway-fabric/issues/1398) to allow larger response headers, so it errors.
So basically can't run suite crm `8.6.1` on there until this is resolved. Was working in `8.5.1` (still might have been returning multiple times, but was few enough that the proxy didn't complain)

### Version

8.6.1

### What browser are you currently using?

Chrome

### Environment Information

PHP 8.2

### Operating System and Version

Debian 12

**Comment by chris001** [comment]
Does this also happen [on the suite 8 online demo](https://suite8demo.suiteondemand.com/) ?  User will pass will

**Comment by DisabledMonkey** [comment]
Online demo appears to be working as expected. Not seeing the problem there.

**Comment by chris001** [comment]
It looks like [a similar Repeated Headers issue was fixed in 2022 in Nginx Gateway Fabric](https://github.com/nginxinc/nginx-gateway-fabric/pull/133).
Could you possibly [troubleshoot whether the repeated session headers are coming at the Nginx Gateway Fabric](https://github.com/nginxinc/nginx-gateway-fabric?tab=readme-ov-file#contacts)?
The simplest way might be to [re-post your original post from here as a new issue on the NGF github](https://github.com/nginxinc/nginx-gateway-fabric/issues/new?assignees=&labels=&projects=&template=bug_report.md&title=) and ask for input from users there.
Please post back here with replies you receive, if any.

**Comment by chris001** [comment]
Nice!  It looks like you found the cause of this issue.

**Comment by DisabledMonkey** [comment]
To get stuff working as expected across the board with the code change above, I also had to swap out any places in code that had `LEGACYSESSID` to be `PHPSESSID` to make everywhere in code reference just that single session name.

So all this to say, I do feel like something weird is going on with suite crm's session juggling there that results in this problem, but don't feel like my code changes are necessarily the appropriate way to fix the underlying problem.

I am curious as to how the demo server is configured as it doesn't seem to present this problem?

**Comment by johnM2401** [comment]
Hey Folks!

Thank you for raising and detailing this issue.

With the release of 8.7.0, SuiteCRM8 moved from the combined PHPSESSID/LEGACYSESSID into using just using one: SCRMSESSID
https://docs.suitecrm.com/8.x/admin/releases/8.7/

I also seem to only have a few set-cookie response headers now, per request:

![Image](https://github.com/user-attachments/assets/87188138-8f70-430e-99f1-c1edfd0b4f5e)

<br>

As this _should_ resolve your issue with swapping sessions often, I'm going to close this issue off for now.

However, if you're still seeing this issue on more recent versions of SuiteCRM8, please don't hesitate to get in touch and we'd be happy to investigate further.

Thank you to both of you for your time and contributions on this!
